cmake/Modules/Packages/USER-COLVARS.cmake

```diff
@@ -12,6 +12,8 @@ if(COLVARS_LEPTON)
   if(NOT BUILD_SHARED_LIBS)
     install(TARGETS lepton EXPORT LAMMPS_Targets LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif()
+  # Change the define below to LEPTON_BUILDING_SHARED_LIBRARY when linking Lepton as a DLL with MSVC
```

Comment:
Could you do something like `if(NOT MSVC)` instead of a comment?

Reply:
Good idea. I assume that the `MSVC` flag is consistent across CMake versions... (*ahem*).

But the right condition to check should be whether or not the ancillary libraries used by LAMMPS are built and linked dynamically. Is there a condition where this may happen, and what would be the flag that expresses it?  With `-DBUILD_SHARED_LIBS=on` I get a LAMMPS shared object as expected, but all the libraries used to build it are static (`.a`).

Reply:
@giacomofiorin in LAMMPS all "built" libraries will always be built as "static" libraries, i.e. libsome_name.a, even when `BUILD_SHARED_LIBS=on`. Only "external" libraries, e.g. FFTW, jpeg, png, MPI may be linked as shared libs. Thus, as far as the USER-COLVARS package is concerned, the colvars and lepton libs will be static libs (just compiled with -fPIC, if needed).

Reply:
we just merged a PR trying to unify and simplify the building of executable and library with the long-term goal to have to have liblammps.so (or .dylib, or .dll) and liblammps.a have the same contents. and the executable being linked from either as determined by the static vs. shared selection.

Reply:
yes it would. adding you.